### ISSUE TITLE
Editor crashes when I copy or duplicate a specific instanced scene

### ISSUE THREAD

**Issue by hmturnbull** [godotengine/godot#74406]
### Godot version

v4.0.stable.official [92bee43ad]

### System information

Windows 10

### Issue description

Godot crashes whenever I try to copy or duplicate instances of a specific scene.  The crash is instant and doesn't even leave a log.  I've tried copying and duplicating other instanced scenes, but those seem to work just fine.

One possible factor is that the affected project was originally one I was using to test out some features in Godot 4 Beta; maybe something was corrupted back then and I only noticed it now?

### Steps to reproduce

1. Open the project.
2. Open the world scene at `res://world/world_01.tscn`.
3. Select the `EnemyTest` node in the scene tree.
4. Try to either copy or duplicate the node (it crashes whether I use a keyboard shortcut or the context menu)
5. The editor will have crashed.

### Minimal reproduction project

[23-03-04 _ Duplication Crash.zip](https://github.com/godotengine/godot/files/10890455/23-03-04._.Duplication.Crash.zip)

(I *think* I removed all the unnecessary files)

**Comment by KoBeWi** [comment]
Crash log
```
CrashHandlerException: Program crashed
Engine version: Godot Engine v4.1.dev.custom_build (60fefcf8527ed57c82d957c480871fe7fd6a11f3)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[0] Node::get_node_or_null (C:\godot_source\scene\main\node.cpp:1272)
[1] Node::get_node (C:\godot_source\scene\main\node.cpp:1346)
[2] Node::_duplicate_signals (C:\godot_source\scene\main\node.cpp:2406)
[3] Node::duplicate_from_editor (C:\godot_source\scene\main\node.cpp:2333)
[4] Node::duplicate_from_editor (C:\godot_source\scene\main\node.cpp:2317)
[5] SceneTreeDock::_tool_selected (C:\godot_source\editor\scene_tree_dock.cpp:681)
[6] SceneTreeDock::shortcut_input (C:\godot_source\editor\scene_tree_dock.cpp:126)
[7] Node::_call_shortcut_input (C:\godot_source\scene\main\node.cpp:2780)
[8] SceneTree::_call_input_pause (C:\godot_source\scene\main\scene_tree.cpp:958)
[9] Viewport::push_unhandled_input (C:\godot_source\scene\main\viewport.cpp:2869)
[10] Window::_window_input (C:\godot_source\scene\main\window.cpp:1375)
[11] call_with_variant_args_helper<Window,Ref<InputEvent> const &,0> (C:\godot_source\core\variant\binder_common.h:293)
[12] call_with_variant_args<Window,Ref<InputEvent> const &> (C:\godot_source\core\variant\binder_common.h:408)
[13] CallableCustomMethodPointer<Window,Ref<InputEvent> const &>::call (C:\godot_source\core\object\callable_method_pointer.h:105)
[14] Callable::callp (C:\godot_source\core\variant\callable.cpp:51)
[15] DisplayServerWindows::_dispatch_input_event (C:\godot_source\platform\windows\display_server_windows.cpp:2316)
[16] DisplayServerWindows::_dispatch_input_events (C:\godot_source\platform\windows\display_server_windows.cpp:2281)
[17] Input::_parse_input_event_impl (C:\godot_source\core\input\input.cpp:690)
[18] Input::flush_buffered_events (C:\godot_source\core\input\input.cpp:941)
[19] DisplayServerWindows::process_events (C:\godot_source\platform\windows\display_server_windows.cpp:2010)
[20] OS_Windows::run (C:\godot_source\platform\windows\os_windows.cpp:1295)
[21] widechar_main (C:\godot_source\platform\windows\godot_windows.cpp:181)
[22] _main (C:\godot_source\platform\windows\godot_windows.cpp:203)
[23] main (C:\godot_source\platform\windows\godot_windows.cpp:217)
[24] WinMain (C:\godot_source\platform\windows\godot_windows.cpp:231)
[25] __scrt_common_main_seh (D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
[26] <couldn't map PC to fn name>
-- END OF BACKTRACE --
```
Caused by this
https://github.com/godotengine/godot/blob/61d2c855114c824f5ca27ded0a1fa71cc7b21134/scene/main/node.cpp#L2151-L2152
`get_class()` returns SplitContainerDragger, which can't be instantiated.

The problem comes from HSplitContainer in HealthBar scene. What's interesting, duplicating the HSplitContainer itself doesn't give any error, nor does duplicating the HealthBar. Only the enemy test has this problem.
The hierarchy is roughly this
- World
  - EnemyTest (instance) <- duplicating this crashes
    - HealthBar (instance)
      - HSplit <- this causes crash

While the direct cause of the crash is caused by SplitContainerDragger not being properly registered, I think it's worth investigating why only the parent instance causes the crash.

**Comment by hmturnbull** [comment]
> While the direct cause of the crash is caused by SplitContainerDragger not being properly registered, I think it's worth investigating why only the parent instance causes the crash.

Fascinating.

Well, many thanks for helping me solve my immediate problem.  Hopefully my report will be of some use in the hunt for bugs.

Cheers.

**Comment by IceflowRE** [comment]
Might be only affecting the root node. Having
- Control
  - HSplitContainer

or 
- Control
  - Control
    - HSplitContainer

Will lead to an error (**no** crash, but with another arrangement i have a crash too).
```
ERROR: Class 'SplitContainerDragger' or its base class cannot be instantiated.
   at: (core/object/class_db.cpp:341)
ERROR: Parameter "obj" is null.
   at: _duplicate (scene/main/node.cpp:2443)
ERROR: Condition "!dup" is true. Continuing.
   at: _tool_selected (editor/scene_tree_dock.cpp:486)
```

**Comment by 14bird** [comment]
I meet the same problem. I use Windows11 and godot version is 4.1.1-stable_mono and recurrent in 4.1.2-stable.
This is my node example:
control
-vboxcontainer
--hsplitcontainer
The control node has a script and vboxcontainer has a signal connect in control's script.
When I copy the root node control, the godot crash and exit, in console has the same output.

ERROR: Class 'SplitContainerDragger' or its base class cannot be instantiated.
   at: (core/object/class_db.cpp:341)
ERROR: Parameter "obj" is null.
   at: _duplicate (scene/main/node.cpp:2443)

**Comment by bhottovy** [comment]
I have just started to encounter this same problem in 4.3dev2 mono.
If I copy/duplicate any scene in the editor that has a SplitContainer/VSplitContainer/HSplitContainer as a child I get the output:
![image](https://github.com/godotengine/godot/assets/17091612/078b2456-90d2-4f55-b59c-911724492e15)
It doesn't seem to matter what the parent's node type is it still has the same result.

**Comment by MoltenCoreDev** [comment]
I compiled the 4.3 dev version precisely commit [4e990cd](https://github.com/godotengine/godot/commit/4e990cd7e51d17cf24f854cc33b2715eaa27200f), the issue seems to be fixed there. I checked twice going back and forth, godot 4.2 crashes, godot 4.3 does not.

**Comment by KoBeWi** [comment]
Fixed by #84824
